pandas/tests/indexing/test_indexing.py

```diff
@@ -92,8 +98,7 @@ def test_getitem_ndarray_3d(self, index, frame_or_series, indexer_sli):
 
         potential_errors = (IndexError, ValueError, NotImplementedError)
         with pytest.raises(potential_errors, match=msg):
-            with tm.assert_produces_warning(DeprecationWarning):
```

Comment:
why did this change?

Reply:
The DeprecationWarning was added in https://github.com/pandas-dev/pandas/pull/30588, but in the meantime that warning has been changed to a FutureWarning, so this test is wrong in any case ..

Reply:
It appears that nesting an `assert_produces_warning` inside a `pytest.raises` simply doesn't work (tried it with a dummy example, and once the code also raises an exception, it doesn't matter whether or which warning is raised, it always passes).

The reason I initially ran into the error, then, is because I first had a case that didn't raise an error, in which case it's first the `assert_produces_warning` which notices that no warning is raised, before the `pytest.raises` notices that no error is raised.

Reply:
That doesn't seem to involve pytest.raises?

Reply:
gotcha, separate issue i guess